test/unit/util/titled-hoc.test.jsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import configureStore from 'redux-mock-store';
+import {mountWithIntl} from '../../helpers/intl-helpers.jsx';
+
+import TitledHOC from '../../../src/lib/titled-hoc.jsx';
+
+jest.mock('react-ga');
+
+describe('TitledHOC', () => {
+    const mockStore = configureStore();
+    let store;
+
+    beforeEach(() => {
+        store = mockStore({
+            scratchGui: {
+                projectState: {}
+            }
+        });
+    });
+
+    test('when a new title is set within GUI, it calls onUpdateProjectTitle', () => {
+        const Component = () => (<div />);
+        const WrappedComponent = TitledHOC(Component);
+        const mockOnUpdateProjectTitle = jest.fn();
+        const mounted = mountWithIntl(
+            <WrappedComponent
+                onUpdateProjectTitle={mockOnUpdateProjectTitle}
+                projectTitle="existing title"
+                store={store}
+            />
+        );
+        expect(mockOnUpdateProjectTitle).not.toHaveBeenCalled();
+        mounted.setProps({
+            reduxProjectTitle: 'new title'
+        });
+        expect(mockOnUpdateProjectTitle).toHaveBeenCalled();
+        expect(mockOnUpdateProjectTitle.mock.calls[0][0]).toBe('new title');
+    });
+
+    test('when new title is set in GUI but external title is blank, it does not call onUpdateProjectTitle', () => {
```

Comment:
Since we allow blank titles to be set, then I think this behavior would make it impossible to set your title to anything after you did set it to be blank.

Reply:
I agree, thanks for pointing this out. Take a look now.